dlt/common/time.py

```diff
@@ -114,13 +88,72 @@ def ensure_pendulum_datetime(value: TAnyDateTime) -> pendulum.DateTime:
         return ret.in_tz(UTC)
     elif isinstance(value, datetime.date):
         return pendulum.datetime(value.year, value.month, value.day, tz=UTC)
-    elif isinstance(value, str):
-        result = parse_iso_like_datetime(value)
+    elif isinstance(value, (int, float, str)):
+        result = _datetime_from_ts_or_iso(value)
+        if isinstance(result, datetime.time):
+            raise ValueError(f"Cannot coerce {value} to a pendulum.DateTime object.")
         if isinstance(result, pendulum.DateTime):
             return result.in_tz(UTC)
         return pendulum.datetime(result.year, result.month, result.day, tz=UTC)
     raise TypeError(f"Cannot coerce {value} to a pendulum.DateTime object.")
 
 
+def ensure_pendulum_time(value: Union[TAnyDateTime, datetime.time]) -> pendulum.Time:
+    """Coerce a date/time value to a `pendulum.Time` object.
+
+    Args:
+        value: The value to coerce. Can be a pendulum.DateTime, pendulum.Date, datetime, date, time, or iso date/time str.
+
+    Returns:
+        A pendulum.Time object
+    """
+
+    if isinstance(value, datetime.time):
+        if isinstance(value, pendulum.Time):
+            return value
+        return pendulum.time(value.hour, value.minute, value.second, value.microsecond)
+    elif isinstance(value, datetime.datetime):
```

Comment:
is there any reason to extract time from date-time types? my take is to be more strict and accept only `str` which we can parse to a valid time and that's it.